File: qwadaf/teste.hs
somaAte :: Int -> Int
somaAte n = sum [1..n]

main :: IO ()
main = do
    putStrLn "Digite um número:"
    input <- getLine
    let n = read input :: Int
    putStrLn ("Resultado: " ++ show (somaAte n))
    

File: qwadaf/teste.ex
defmodule Soma do
  def soma_ate(n) do
    Enum.sum(1..n)
  end
end

IO.puts "Digite um número:"
n = IO.gets("") |> String.trim() |> String.to_integer()

IO.puts "Resultado: #{Soma.soma_ate(n)}"